src/plot.js

```diff
@@ -96,12 +95,38 @@ export function plot(options = {}) {
       .call(applyInlineStyles, style)
     .node();
 
-  for (const mark of marks) {
+  // A mark layout is responsible for:
+  // * applying the scales, possibly pushing a new one (for legends)
+  // * returning a filtered index and values to render, possibly new channels to rescale
+  const newScaleChannels = new Map();
+  const layouts = [];
+  const newOptions = {};
+  for (let i = 0; i < marks.length; ++i) {
+    const mark = marks[i];
+    const {channels} = layouts[i] = applyLayout.call(mark, mark.layout, [markIndex.get(mark)], scaleDescriptors, markChannels.get(mark), dimensions, options);
```

Comment:
It looks like this is ignoring the *index* and *values* returned by *mark*.layout. I think I follow how the *values* are being ignored because the code currently relies on mutation (https://github.com/observablehq/plot/pull/775/files#r818901600). I’m a little surprised that it’s ignoring the *index* though, here. We don’t ignore the *index* in the faceting case.

Reply:
Oh, never mind. The *layout* is stored in *layouts* and then *layout*.index is referenced later.